### ISSUE TITLE
Internet is too slow in many times with Qualcomm Atheros QCA9377 802.11ac Wireless Network Adapter when I using pop OS 20.04.

### ISSUE THREAD

**Issue by giraffe96** [pop-os/pop#1470]
**Distribution (run `cat /etc/os-release`):**Pop!_OS 20.04 LTS
 
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):** Internet speed and wifi

**Issue/Bug Description:**
I have used Manjaro linux and Linux lite(ubuntu 20.04 lts based) earlier but I don't have any issues with internet connection and wifi too but I have recently switch to pop os because sometimes wifi is not connected when I restart (but this is not often) and major issue is I can able to use the internet with full speed at the beginning(initially(20Mbps)) after sometime(after using 30-45 minutes) net speed is getting extremely slow(like 14kbps-8kbps) [I have noticed this always like more than 15-20 times]. My machine is Dell G3 15 3590 configured with Qualcomm Atheros QCA9377 802.11ac, 16 GB ram and Intel® Core™ i5-9300H CPU( https://www.dell.com/support/home/en-us/product-support/servicetag/0-YUFGdHU4cTJoc3BCVU5iMldVRkxVUT090/overview). I like to mention I have the same issue with official ubuntu 20.04.1 lts but linux lite they have same ubuntu inside that but I haven't face any internet issues with linux lite. I can able make sure my internet connection speed always same because my macbook doesn't have any issues.

**Steps to reproduce (if you know):**
(I am not sure about this) I guess it could happen because driver issues or something else

**Expected behavior:**
I like to use my internet without any interruption on my machine( like Linux lite )

**Other Notes:**

**Comment by leviport** [comment]
The title alone isn't really a sufficient amount of information for this to be a useful bug report. If you fill in the issue template, let me know and I can reopen the issue afterwards. Some useful info would be things like whether or not you see the same behavior in other distros or other OSes, describing the hardware you're running (more than just the wireless card), whether this happens with multiple access points or if you've only tested one, and a description of the speed drops you're seeing (even better if you can get actual speed numbers).

**Comment by giraffe96** [comment]
@leviport I have updated my  bug report. and you can check and let me know if you have any questions regarding this. Thank you @leviport to responce this quickly if you can able to fix this issues would be better for me because I like to use pop os because has better gnome

**Comment by leviport** [comment]
Thank you for elaborating, that's a good amount of information. Since you're also seeing the issue in Ubuntu, I'm suspecting an upstream package in combination with your hardware might be where the bug is coming from. I'm reopening this issue so we can continue to monitor the bug, but I'm not sure that I have access to similar hardware to do any further testing on my end.

**Comment by giraffe96** [comment]
Thank you so much @leviport for your support ! if the pop os is more stable I could encourage to use pop os instead of clean Ubuntu with my colleges.

**Comment by colombin** [comment]
I have the same issue with the Qualcomm Atheros QCA9377 Wifi-Card. I run Pop-os 20.10

**Comment by mind-overflow** [comment]
having the same issue since 20.04, up to 20.10, with a Qualcomm Atheros QCA9377.

**Comment by BaronBaleron** [comment]
I have the same issue on Ubuntu 20.04 with Qualcomm Atheros QCA9377

**Comment by MDCoelho** [comment]
I have the same problem here with Pop_OS 20.04 and Uqalcom Atheros QCA9377 rev 31.

I'm not an expert but I tested with a external tp-link TL-WN722N and for some hours I couldn't reproduce the problem.

**Comment by avdzm** [comment]
Hello all,
I too have similar problem.

I'm having trouble transferring files over wifi.
It is ridiculously slow 1Mb/s.

I have a dell 5575 with a card QCA9377 802.11ac Wireless Network Adapter.
I checked online this card has a Peak Speed: 433 Mbps.

I've attached a screenshot of the network settings (GUI) it is on 1 Mb/s (2.4 GHz).
![network gui settings](https://user-images.githubusercontent.com/64041008/118410752-62db8980-b691-11eb-93ee-2c4c2ed53cc2.png)

I tried commands
sudo iwconfig wlp3s0 rate 50M
disabling ipv6.
rebooting the router, the nas drive, the laptop.

None of the above worked.

A few days ago the speed was fine, I'm not sure if a recent update caused this lag for me.

This is my output for command lshw -C network
description: Wireless interface
       product: QCA9377 802.11ac Wireless Network Adapter
       vendor: Qualcomm Atheros
       physical id: 0
       bus info: pci@0000:03:00.0
       logical name: wlp3s0
       version: 31
       serial: 48:5f:99:54:e9:81
       width: 64 bits
       clock: 33MHz
       capabilities: bus_master cap_list ethernet physical wireless
       configuration: broadcast=yes driver=ath10k_pci driverversion=5.12.4-xanmod1 firmware=WLAN.TF.2.1-00021-QCARMSWP-1 ip=192.168.100.19 latency=0 link=yes multicast=yes wireless=IEEE 802.11
       resources: irq:76 memory:e0200000-e03fffff

I'm happy to provide with additional information.

**Comment by rthiru96** [comment]
can anyone please report #1470 issue to ubuntu  @avdzm  @MDCoelho  @BaronBaleron  @mind-overflow @colombin ?? I guess  they could fix with their new release.

**Comment by mmstick** [comment]
Are you able to get a better connection with the 21.04 beta ISO?

**Comment by brukidm** [comment]
I believe this worked for me, as found on the itsfoss article telling you how to speed up your WiFi:

`
sudo su
echo "options ath9k nohwcrypt=1" >> /etc/modprobe.d/ath9k.conf
`

except it took me a while to realize which ath I should actually use, so after trying `ath10k` and `ath10k_pci` out, the latter made it work.

**Comment by brukidm** [comment]
> Hello all,
> I too have similar problem.
> 
> I'm having trouble transferring files over wifi.
> It is ridiculously slow 1Mb/s.
> 
> I have a dell 5575 with a card QCA9377 802.11ac Wireless Network Adapter.
> I checked online this card has a Peak Speed: 433 Mbps.
> 
> I've attached a screenshot of the network settings (GUI) it is on 1 Mb/s (2.4 GHz).
> ![network gui settings](https://user-images.githubusercontent.com/64041008/118410752-62db8980-b691-11eb-93ee-2c4c2ed53cc2.png)
> 
> I tried commands
> sudo iwconfig wlp3s0 rate 50M
> disabling ipv6.
> rebooting the router, the nas drive, the laptop.
> 
> None of the above worked.
> 
> A few days ago the speed was fine, I'm not sure if a recent update caused this lag for me.
> 
> This is my output for command lshw -C network
> 
> ```
> description: Wireless interface
>        product: QCA9377 802.11ac Wireless Network Adapter
>        vendor: Qualcomm Atheros
>        physical id: 0
>        bus info: pci@0000:03:00.0
>        logical name: wlp3s0
>        version: 31
>        serial: 48:5f:99:54:e9:81
>        width: 64 bits
>        clock: 33MHz
>        capabilities: bus_master cap_list ethernet physical wireless
>        configuration: broadcast=yes driver=ath10k_pci driverversion=5.12.4-xanmod1 firmware=WLAN.TF.2.1-00021-QCARMSWP-1 ip=192.168.100.19 latency=0 link=yes multicast=yes wireless=IEEE 802.11
>        resources: irq:76 memory:e0200000-e03fffff
> ```
> 
> I'm happy to provide with additional information.
> 
> I've also attached a photo of the transfer speed.
> ![transfer speed](https://user-images.githubusercontent.com/64041008/118411180-8e5f7380-b693-11eb-9d30-e1706b3690b1.png)

Have you tried changing `wifi.powersave` in `/etc/NetworkManager/conf.d/default-wifi-powersave-on.conf` from 3 to 2?

**Comment by michaelgrigoryan25** [comment]
Experiencing the same issue here with the same driver and Pop_OS version `20.04 LTS`.
I am expecting about 50mb/s from my __5GHz__ connection.
After changing the value from `3` to `2` made a huge difference(10mb/s to be exact) for my __2.4GHz__ connection @brukidm

**Comment by brukidm** [comment]
@michaelgrigoryan25 what helped me and what I hope wasn't a random hiccup was disabling 2.4GHz network and leaving only 5GHz active. You can try it out for the sake of the tests and keep it anyway if all your device (can) use 5GHz.

**Comment by michaelgrigoryan25** [comment]
> What helped me and what I hope wasn't a random hiccup was disabling 2.4GHz network and leaving only 5GHz active.

I don't think that would be a working solution, since I get 22mb/s on my phone and the speed is fine for everybody except me.

> I don't think our adapter is impacted by ath9k, but rather ath10k and ath10k_pci. Unfortunately, the command I used in the above comment is not recognized, as I saw later in logs.

Oh. That's unfortunate, should I try switching to Ubuntu then? As I remember I didn't run into such issue when I tried it out.

**Comment by brukidm** [comment]
@michaelgrigoryan25 what I can suggest is that you try a live Ubuntu USB and see if it's better, although Pop is Ubuntu based so I doubt it will make any difference.

**Comment by michaelgrigoryan25** [comment]
Quick update @brukidm.

I have deleted the file at `/lib/firmware/ath10k/QCA9377/hw1.0/firmware-6.bin` and rebooted my system. Now both links work as expected.

Here are my results after removing it.

|2.4GHz|5GHz
-|-
20mb/s|40mb/s

[link to the answer given on AskUbuntu](https://askubuntu.com/a/1312339)

**Comment by michaelgrigoryan25** [comment]
Any updates on this issue?

**Comment by UjjwalSinghKhunte** [comment]
I too have same issue in linux mint
But,
Running below command seems to fix issue but issue reappears after restartng system and running this command again fixes issue temporarily.
$ sudo modprobe -r ath10k_pci
$ sudo modprobe ath10k_pci

**Comment by michaelgrigoryan25** [comment]
Pop!_OS: `21.10`
Nvidia driver: None

Tested using VMware Workstation 16

Seems like in the VM it was able to max out the speed automatically(`1000 mb/s`. I suppose that the reading comes from the virtual VMware driver) with "wired" connection.

![Settings > Network > Wired connection](https://user-images.githubusercontent.com/56165400/147326470-398143a9-0f37-4043-b3ca-d0086516a871.png)

When I installed Pop by flashing the whole disk the shown bandwidth was I think `1mb/s` or something like that. Could the invalid reading potentially be the issue?

###### I am using NAT(sharing the host's IP address with the VM) and getting the desired speeds from https://speedtest.net/ on both links.

**Comment by brukidm** [comment]
> Finally managed to test this out in a VM. Here are some details.
> 
> Pop!_OS: `21.10` Nvidia driver: None
> 
> Tested using VMware Workstation 16
> 
> Seems like in the VM it was able to max out the speed automatically(`1000 mb/s`. I suppose that the reading comes from the virtual VMware driver) with "wired" connection.
> 
> ![Settings > Network > Wired connection](https://user-images.githubusercontent.com/56165400/147326470-398143a9-0f37-4043-b3ca-d0086516a871.png)
> 
> When I installed Pop by flashing the whole disk the shown bandwidth was I think `1mb/s` or something like that. Could the invalid reading potentially be the issue?
> ###### I am using NAT(sharing the host's IP address with the VM) and getting the desired speeds from https://speedtest.net/ on both links.

I'm pretty sure VM uses Windows' drivers, especially if VM interprets this as a wired connection, so I wouldn't take VM test case into consideration here.

**Comment by michaelgrigoryan25** [comment]
> I'm pretty sure VM uses Windows' drivers, especially if VM interprets this as a wired connection, so I wouldn't take VM test case into consideration here.

You are absolutely correct. I forgot that network was basically being "bridged". That was a mistake on my side. I will try to see whether there is a way to use physical drivers to test this out. Thanks for mentioning this @brukidm

**Comment by mind-overflow** [comment]
Hey guys, I don't have a way to help you but I have an update.  
I was having this issue in my house, where I have a 200 Mb/s connection but my laptop wouldn't go over 30 on Linux (Windows was fine). Today, I went to my University and ran a speed test with their connection - and it got up to about 100 Mb/s!  
So, my guess is that this has to do with how the driver handles different auth protocols, bands, or frequencies - my Uni is using WPA2 Enterprise with a private-key authentication, and probably their network is configured completely differently from mine (channel bandwidth, 5/2.4 GHz, b/g/n...)  
Since I have installed OpenWRT at home, I'll try to change some things up and see if I can get any improvement.

**Comment by mmstick** [comment]
You could try disabling channel bonding in your router. Try different channel widths for 2.4 and 5 GHz access points, starting with 20 MHz.

**Comment by michaelgrigoryan25** [comment]
I tried actually sometime ago, but that didn't change anything though 🤔

**Comment by Shakhawat001** [comment]
I tried many things but it's the same. I think It's a driver problem.

`iwconfig`
```
lo        no wireless extensions.

enp2s0    no wireless extensions.

wlp1s0    IEEE 802.11  ESSID:"mynetwork"  
          Mode:Managed  Frequency:2.412 GHz  Access Point: Router_mac
          Bit Rate=1 Mb/s   Tx-Power=20 dBm   
          Retry short limit:7   RTS thr:off   Fragment thr:off
          Power Management:off
          Link Quality=34/70  Signal level=-76 dBm  
          Rx invalid nwid:0  Rx invalid crypt:0  Rx invalid frag:0
          Tx excessive retries:0  Invalid misc:25   Missed beacon:0

```

**Comment by rsJuuuuu** [comment]
To add to UjjwalSinghKhunte's solution above. I observed that after modprobe p2p is missing from nmcli output. Therefore, I figured it had something to do with the issue. As far as I know it is not possible to disable p2p wifi, while using NetworkManager [https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/537](https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/issues/537). So instead I just built wpasupplicant from source without the p2p enabled. For me this resolved the issue for good. 

Obviously you lose the p2p functionality doing this so it is not a fix for everyone. Now, I have not debugged any further, so there might be a more specific feature that causes the problem and as such you might be able to find a more specific fix that would allow one to keep most or all of the p2p functionality. I guess ideally NetworkManager would just support disabling p2p and we could simply flip a switch to get full wifi speeds.

**Comment by MahendraVirakthmath** [comment]
I tried to do what @michaelgrigoryan25 did. I deleted the file /lib/firmware/ath10k/QCA9377/hw1.0/firmware-6.bin by keeping a backup of it (Yeah! I did restart), but it did not work for me. Then I made it to specifically use firmware-5.bin and it worked. The script that I use is:

#!/bin/bash
nmcli radio wifi off
modprobe -r ath10k_pci
modprobe ath10k_pci firmware=firmware-5.bin
nmcli radio wifi on

I am not sure if you need to turn wifi off before you unload and load your wifi module with the firmware-5.bin but I just feel like to do it

Instead of running this every time you boot, you can just make it run at startup and you are good to go

PS: Strangely by doing this it doubled my wifi speed. I have a 100Mbps connection and I was getting around 20+Mbps speed after doing this workaround it got upto 50+Mbps

**Comment by FlyingSaturn** [comment]
I'm having the same issue. I use KDE Neon and yeah, it's an upstream issue but I did notice it on Pop!_OS.

**Comment by FlyingSaturn** [comment]
I have tried to provide some of the solutions in a Reddit post: https://www.reddit.com/r/linux/s/egXvRBYhYp

This isn't a foolproof solution though.
